sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/DataLakeTestBase.java

```diff
@@ -419,6 +403,28 @@ protected DataLakeDirectoryClient getDirectoryClient(StorageSharedKeyCredential
             .buildDirectoryClient();
     }
 
+    protected DataLakePathClientBuilder setOauthCredentials(DataLakePathClientBuilder builder) {
+        if (ENVIRONMENT.getTestMode() != TestMode.PLAYBACK) {
+            // AZURE_TENANT_ID, AZURE_CLIENT_ID, AZURE_CLIENT_SECRET
+            return builder.credential(new EnvironmentCredentialBuilder().build());
+        } else {
+            // Running in playback, we don't have access to the AAD environment variables, just use SharedKeyCredential.
+            return builder.credential(ENVIRONMENT.getPrimaryAccount().getCredential());
```

Comment:
Should make a separate helper method to get a TokenCredential as this should use that same logic as I commented on above.